Model/Sync/Reset/Catalog.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace Yotpo\Core\Model\Sync\Reset;
+
+use Magento\Framework\App\ResourceConnection;
+use Yotpo\Core\Model\AbstractJobs;
+use Yotpo\Core\Model\Config as CoreConfig;
+use Yotpo\Core\Model\Sync\Data\Main as SyncDataMain;
+
+class Catalog extends Main
+{
+    const PRODUCT_SYNC_TABLE = 'yotpo_product_sync';
+    const CATEGORY_SYNC_TABLE = 'yotpo_category_sync';
+    const CRONJOB_CODES = ['yotpo_cron_core_category_sync','yotpo_cron_core_products_sync'];
+
+    /**
+     * @var SyncDataMain
+     */
+    protected $syncDataMain;
+
+    /**
+     * @param ResourceConnection $resourceConnection
+     * @param CoreConfig $coreConfig
+     * @param SyncDataMain $syncDataMain
+     * @param AbstractJobs $abstractJobs
+     */
+    public function __construct(
+        ResourceConnection $resourceConnection,
+        CoreConfig $coreConfig,
+        SyncDataMain $syncDataMain,
+        AbstractJobs $abstractJobs
+    ) {
+        parent::__construct(
+            $resourceConnection,
+            $coreConfig,
+            $abstractJobs
+        );
+        $this->syncDataMain = $syncDataMain;
+    }
+
+    /**
+     * @param int $storeId
+     * @return void
+     */
+    public function resetSync($storeId)
+    {
+        $this->setStoreId($storeId);
+        $this->setCronJobCodes(self::CRONJOB_CODES);
+        parent::resetSync($storeId);
+        $catalogTables = [self::PRODUCT_SYNC_TABLE, self::CATEGORY_SYNC_TABLE];
+        foreach ($catalogTables as $table) {
+            $tableName = $this->resourceConnection->getTableName($table);
+            $totalCount = $this->getTotalCount($tableName, $storeId);
+            $this->deleteFromTable($tableName, $totalCount, $storeId);
```

Comment:
Please move the `getTotalCount` into the deletion method and rename it to `deleteAllFromTable`
```suggestion
            $this->deleteAllFromTable($tableName, $totalCount, $storeId);
```

Reply:
This is done. Will update the PR soon